sockeye/inference.py

```diff
@@ -598,19 +610,20 @@ def make_input_from_json_string(sentence_id: int, json_string: str) -> Translato
     :param json_string: A JSON object serialized as a string.
     :return: A TranslatorInput.
     """
-    jobj = json.loads(json_string, encoding=C.JSON_ENCODING)
-    tokens = jobj.get(C.JSON_TEXT_KEY)
-    if tokens is None:
-        raise ValueError("Missing 'text' field in JSON string: %s", json_string)
-
-    tokens = list(data_io.get_tokens(tokens))
-    factors = jobj.get(C.JSON_FACTORS_KEY)
-    if isinstance(factors, list):
-        factors = [list(data_io.get_tokens(factor)) for factor in factors]
-    else:
-        factors = None
+    try:
+        jobj = json.loads(json_string, encoding=C.JSON_ENCODING)
+        tokens = jobj[C.JSON_TEXT_KEY]
+        tokens = list(data_io.get_tokens(tokens))
+        factors = jobj.get(C.JSON_FACTORS_KEY)
+        if isinstance(factors, list):
+            factors = [list(data_io.get_tokens(factor)) for factor in factors]
+        else:
+            factors = None
+        return TranslatorInput(sentence_id=sentence_id, tokens=tokens, factors=factors)
 
-    return TranslatorInput(sentence_id=sentence_id, tokens=tokens, factors=factors)
+    except Exception as e:
+        logger.exception(e, exc_info=True) if not is_python34() else logger.error(e)  # type: ignore
```

Comment:
this is what I was referring to: here the string is not logged.

Reply:
changed